### PR TITLE
Feature/dropdown menu

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/RawButton/index.tsx [hunk 1]
```diff
@@ -7,21 +7,21 @@ import { genericMemo } from '../../utils';
...
+export interface Props<N extends number | string> extends Omit<React.HTMLProps<HTMLButtonElement>, 'ref' | 'onClick' | 'name'>{
     /**
     * Style for the button
     */
     className?: string;
     /**
     * Gets called when user clicks on the button
     */
-    onClick?: (name: N, e: React.MouseEvent<HTMLButtonElement>) => void;
+    onClick?: (name: N | undefined, e: React.MouseEvent<HTMLButtonElement>) => void;
```
Comment (tnagorra) [inline src/components/RawButton/index.tsx]: This will break typings

**Review by tnagorra — CHANGES_REQUESTED** [review]
We can't do that

File: src/components/InputContainer/index.tsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ export interface Props {
     uiMode?: UiMode;
     filled?: boolean;
     children?: React.ReactNode;
+    variant: 'form' | 'general';
```
Comment (tnagorra) [inline src/components/InputContainer/index.tsx]: Shouldn't this be optional if there is a default value?

Comment (frozenhelium) [reply src/components/InputContainer/index.tsx]: Good point

File: src/components/RawButton/index.tsx [hunk 3]
```diff
@@ -7,21 +7,18 @@ import { genericMemo } from '../../utils';
 
 import styles from './styles.css';
 
-export interface Props<N extends number | string | undefined> extends Omit<React.HTMLProps<HTMLButtonElement>, 'ref' | 'onClick' | 'name'>{
+interface Props<N extends string | number | undefined> extends Omit<React.HTMLProps<HTMLButtonElement>, 'ref' | 'onClick' | 'name'>{
```
Comment (tnagorra) [inline src/components/RawButton/index.tsx]: why no export this interface any longer?

File: src/components/RawButton/index.tsx [hunk 4]
```diff
@@ -7,21 +7,18 @@ import { genericMemo } from '../../utils';
...
     /**
     * Style for the button
     */
     className?: string;
-    /**
-    * Gets called when user clicks on the button
-    */
```
Comment (tnagorra) [inline src/components/RawButton/index.tsx]: Why remove the comment and re-order the props?

File: src/components/RawButton/index.tsx [hunk 5]
```diff
@@ -34,7 +31,7 @@ export interface Props<N extends number | string | undefined> extends Omit<React
  * we use raw button as a base component for other types of buttons or
  * clickable elements
  */
-function RawButton<N extends number | string | undefined>(props: Props<N>) {
+function RawButton<N extends number | string>(props: Props<N>) {
```
Comment (tnagorra) [inline src/components/RawButton/index.tsx]: Button can extend `undefined`. This would be a breaking change!

**Review by tnagorra — CHANGES_REQUESTED** [review]
Let's add stories so that we can capture the UI snapshots

**Comment by frozenhelium** [comment]
> Let's add stories so that we can capture the UI snapshots

We already have
https://605879dde6ab9e0021de652b-aqirikcxew.chromatic.com/?path=/story/view-dropdownmenu--default

**Review by tnagorra — APPROVED** [review]

File: src/components/DropdownMenuItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            </RawButton>
+        );
+    }
+
+    return (
+        <RawButton<undefined>
+            className={className}
+            name={undefined}
+            // eslint-disable-next-line react/destructuring-assignment
+            onClick={props.onClick as RawButtonProps<undefined>['onClick']}
```
Comment (tnagorra) [inline src/components/DropdownMenuItem/index.tsx:119]: This looks a bit weird.
Where is `name` used if we are just passing `undefined`?

Comment (frozenhelium) [reply src/components/DropdownMenuItem/index.tsx:119]: This is the case wher use doesn't provide any name for the dropdown item

File: src/components/DropdownMenuItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            >
+                {content}
+            </RawButton>
+        );
+    }
+
+    return (
+        <RawButton<undefined>
+            className={className}
+            name={undefined}
```
Comment (frozenhelium) [inline src/components/DropdownMenuItem/index.tsx:117]: @tnagorra  here it is

**Pull request merged**